contract/src/lib.rs

```diff
@@ -290,10 +293,32 @@ impl MpcContract {
     }
 
     #[allow(unused_variables)]
-    pub fn sign(&mut self, payload: [u8; 32]) -> [u8; 32] {
-        near_sdk::env::random_seed_array()
+    pub fn sign(&mut self, payload: [u8; 32]) -> Promise {
+        let hash = env::sha256(&payload);
+        self.pending_requests.insert(&hash, &None);
+        Self::ext(env::current_account_id()).sign_helper(hash)
+    }
+
+    pub fn sign_helper(&mut self, payload_id: Vec<u8>) -> PromiseOrValue<String> {
+        if let Some(signature) = self.pending_requests.get(&payload_id) {
+            match signature {
+                Some(signature) => {
+                    self.pending_requests.remove(&payload_id);
+                    PromiseOrValue::Value(signature)
+                }
+                None => {
+                    env::log_str("not ready");
+                    let account_id = env::current_account_id();
+                    PromiseOrValue::Promise(Self::ext(account_id).sign_helper(payload_id))
```

Comment:
Would it be possible to extend the delay for longer, for cheaper, by chaining 1-yocto transfers before the reflexive call, especially if a minimum delay is expected?

Reply:
How would chaining 1-yotco transfers help?

Reply:
I think the assumption here is that 1-yocto transfer is cheaper than a cross-contract call

Reply:
Maybe I am missing something here but I think you still have to call self again in order to continuously check whether a signature has been generated.

Reply:
Yeah, so I think what @encody means is that we could explore whether it makes sense to chain 10 1-yocto transfers first to create cheap superficial delay since we know that the response is not going to arrive in the first ~10 seconds (or whatever constant makes sense here), and then chain self-calls afterward expecting the response to have potentially come in by then.